### DISCUSSION TITLE
Get pixel witdth and hight

### DISCUSSION THREAD

**Discussion by Peregalli** [FAST-Imaging/FAST#192] [Q&A] [answered]
Hi!

I am using Fast library in python, and importing a WSI with `TIFFImagePyramidImporter` function and after that I am extracting some images with  `getPatchAsImage`, with level 0 (40x magnification) and 1024x1024 image size. I want to see size of some structures, how can I get the pixel witdth and hight in micrometers or milimeters?

**Comment by smistad** [comment, marked as answer]
Hi

If that information is stored in your TIFF you can get that information with the getSpacing() method.

For an extracted patch at level, x, y with a specific width and height:
```python
access = imagePyramid.getAccess(fast.ACCESS_READ)
image = access.getPatchAsImage(level, x, y, width, height)
spacing_x = image.getSpacing()[0,0]
spacing_y = image.getSpacing()[1,0]
```
This corresponds to the spacing for the given **level** of the patch.

You can also get the spacing for level 0 directly from the image pyramid:
```python
spacing_x = imagePyramid.getSpacing()[0,0]
spacing_y = imagePyramid.getSpacing()[1,0]
```

**Reply by Peregalli** [reply]
Thanks for your answer! In my case those commands didnt work, but I was able to extract that information from WSI metadata.

**Reply by andreped** [reply]
>  I was able to extract that information from WSI metadata.

@Peregalli Would be great if you could share exactly what you did (minimalistic code example), such that we and others who might run into the same issue, may find a solution for it by finding this discussion thread :)

**Reply by smistad** [reply]
The first example I wrote was incorrect. I updated my answer.
